staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -59,6 +59,8 @@ const (
 	StaticEstimatedCostLimit = 10000000
 	// StaticEstimatedCRDCostLimit represents the largest-allowed total cost for the x-kubernetes-validations rules of a CRD.
 	StaticEstimatedCRDCostLimit = 100000000
+
+	MaxSelectableFields = 8
```

Comment:
I'm open to suggestions on what the max limit should be.  There is a small memory footprint for each selectable field since selectable field values are extracted from resources in the watch cache, and there is potential to index these in the future (we wouldn't want a limit so high that it is impossible to do that).

Reply:
Beyond alpha, could we make it an API server configuration setting? Different clusters might have different needs.

Reply:
> Beyond alpha, could we make it an API server configuration setting? Different clusters might have different needs.

This would break CRD portability. Could we find high enough value that would work for most use cases? I'm open to something in the 32-64 range..